dxr/static_unhashed/js/dxr.js

```diff
@@ -155,8 +155,13 @@ $(function() {
             searchUrl += '&case=' + isCaseSensitive;
         }
 
-        queryField.val(decodeURIComponent(fromQuery[1]));
+        queryField.val(decodeURIComponent(fromQuery[1].replace('+', ' ')));
         showBubble('info', viewResultsTxt.replace('{{ url }}', searchUrl));
+    } else {
+        var from_qQuery = /[?&]from_q=([^&]+)/.exec(location.search);
```

Comment:
I think we should show a bubble, just to ward off any user surprise. Something along the lines of "There was only 1 search result, so I took you straight there."